statusHandler/termuic/termuiConsole.go

```diff
@@ -0,0 +1,148 @@
+package termuic
+
+import (
+	"github.com/ElrondNetwork/elrond-go/statusHandler/termuic/termuiRenders"
+	"os"
+	"os/signal"
+	"runtime"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	ui "github.com/gizak/termui/v3"
+)
+
+//refreshInterval is used for a ticker that refresh termui console at a specific interval
+const refreshInterval = time.Second
+
+//numLogLines is used to specify how many lines of logs need to store in slice
+const numLogLines = 50
+
+// TermuiConsole data where is store data from handler
+type TermuiConsole struct {
+	termuiConsoleMetrics *sync.Map
+	logLines             []string
+	mutLogLineWrite      sync.RWMutex
+	consoleRender        TermuiRender
+	grid                 *ui.Grid
+}
+
+//NewTermuiConsole method is used to return a new TermuiConsole structure
+func NewTermuiConsole(metricData *sync.Map) *TermuiConsole {
+	tc := TermuiConsole{
+		termuiConsoleMetrics: metricData,
+	}
+
+	return &tc
+}
+
+func (tc *TermuiConsole) Write(p []byte) (n int, err error) {
+	go func() {
+		logLine := string(p)
+		logLine = strings.Replace(logLine, "\n", "", len(logLine))
+		logLine = strings.Replace(logLine, "\r", "", len(logLine))
+
+		tc.mutLogLineWrite.Lock()
+		if len(tc.logLines) >= numLogLines {
+			tc.logLines = tc.logLines[1:numLogLines]
+		}
+		if logLine != "" {
+			tc.logLines = append(tc.logLines, logLine)
+		}
+		tc.mutLogLineWrite.Unlock()
+	}()
+
+	return len(p), nil
+}
+
+// Start method - will start termui console
+func (tc *TermuiConsole) Start() error {
+
+	if err := ui.Init(); err != nil {
+		return err
+	}
+
+	go func() {
+		defer ui.Close()
+
+		tc.eventLoop()
+	}()
+
+	return nil
+}
+
+func (tc *TermuiConsole) eventLoop() {
+
+	tc.grid = ui.NewGrid()
+	tc.consoleRender = termuiRenders.NewWidgetsRender(tc.termuiConsoleMetrics, tc.grid)
+
+	termWidth, termHeight := ui.TerminalDimensions()
+	tc.grid.SetRect(0, 0, termWidth, termHeight)
+
+	time.Sleep(1 * time.Second)
+
+	uiEvents := ui.PollEvents()
+	// handles kill signal sent to gotop
+	sigTerm := make(chan os.Signal, 2)
+	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
+
+	drawTicker := time.NewTicker(refreshInterval).C
+
+	for {
+		select {
+		case <-drawTicker:
+			tc.consoleRender.RefreshData(tc.logLines)
+
+			ui.Render(tc.grid)
+
+		case <-sigTerm:
+			ui.Clear()
+
+			return
+		case e := <-uiEvents:
+			tc.processUiEvents(e)
+		}
+	}
+}
+
+func (tc *TermuiConsole) processUiEvents(e ui.Event) {
+	switch e.ID {
+	case "<Resize>":
+		payload := e.Payload.(ui.Resize)
+		tc.grid.SetRect(0, 0, payload.Width, payload.Height)
+		ui.Clear()
+		ui.Render(tc.grid)
+	case "q":
+		tc.changeConsoleDisplay()
+
+	case "<C-c>":
+		ui.Close()
+		if p, err := os.FindProcess(os.Getpid()); err != nil {
```

Comment:
This should be extracted in some file (processCloser) that will handle different OS/Arch.

Reply:
created a new file with name processCloser where the current process is closed.